jsonschema-generator/src/test/java/com/github/victools/jsonschema/generator/SchemaGeneratorOptionsTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 VicTools.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.victools.jsonschema.generator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import junitparams.naming.TestCaseName;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test for {@link SchemaGenerator} class.
+ */
+@RunWith(JUnitParamsRunner.class)
+public class SchemaGeneratorOptionsTest {
+
+    Object parametersForTestGenerateSchema() {
+        return new Object[][]{
+            {"properties in alphabetic order", null, Arrays.asList(
+                "CONSTANT_A", "CONSTANT_B", "fieldInSupertype", "firstField", "lastField", "middleField",
+                "firstMethod()", "lastMethod()", "methodInSupertype()", "middleMethod()"
+                )},
+            {"properties in declaration order", Option.PROPERTIES_IN_DECLARATION_ORDER, Arrays.asList(
+                "fieldInSupertype", "firstField", "middleField", "lastField", "CONSTANT_B", "CONSTANT_A",
+                "middleMethod()", "firstMethod()", "lastMethod()", "methodInSupertype()"
+                )}
```

Comment:
@udowal this is the order returned by the tests here – unfortunately that is not the declaration order.
Now I remember that I wanted to offer this option as well in the past but didn't manage to convince `classmate` to respect the declaration order.

I'm afraid the change is not as simple as hoped.